lib/charms/layer/kubernetes_common.py

```diff
@@ -933,3 +934,199 @@ def get_secret_password(username):
             if username == secret_user:
                 return password
     return None
+
+
+def get_node_ip():
+    """Determines the preferred NodeIP value for this node."""
+    cidr = cluster_cidr()
+    if not cidr:
+        return None
+    if is_ipv6_preferred(cidr):
+        return get_ingress_address6("kube-control")
+    else:
+        return get_ingress_address("kube-control")
+
+
+def merge_kubelet_extra_config(config, extra_config):
+    """Updates config to include the contents of extra_config. This is done
+    recursively to allow deeply nested dictionaries to be merged.
+
+    This is destructive: it modifies the config dict that is passed in.
+    """
+    for k, extra_config_value in extra_config.items():
+        if isinstance(extra_config_value, dict):
+            config_value = config.setdefault(k, {})
+            merge_kubelet_extra_config(config_value, extra_config_value)
+        else:
+            config[k] = extra_config_value
```

Comment:
Removed [here](https://github.com/charmed-kubernetes/charm-kubernetes-worker/pull/103/files#diff-1837fd4ad476d5c87c628d68b3c461fc5c03d6aeafc928e910c69aa6ae267f85L746-L757) and added in this PR